src/components/Modals/Modal.vue

```diff
@@ -40,13 +44,20 @@ export default {
     return {
       isOnboard: false,
       courseIsAddable: true,
-      isDisabled: false
+      isDisabled: false,
+      leftButton: 'CANCEL',
+      goBack: false
     };
   },
   props: {
     type: String,
     semesterID: Number,
-    currentSemesters: Array
+    currentSemesters: Array,
+    season: String,
+    year: Number
+  },
+  mounted() {
+    this.$root.$on('toggle-left-button', this.toggleLeftButton);
```

Comment:
Ok I know $root might make things more convenient, but is there any way of setting and emitting events in a way that avoids it? Just by emitting them in a path upwards from the current component. Want to avoid stuff like this in case the root component or general hierarchy ever changes

Reply:
So unfortunately, emitting without $root from got me no feedback between `Modal` and `NewCourse` (and I'm assuming maybe it's because it's a modal?). Thoughts on making a [bus](https://blog.logrocket.com/using-event-bus-in-vue-js-to-pass-data-between-components/)?

Reply:
Could be worth a shot! I have never done this before so not sure how well it will go but it could be an improvement! Thanks for the link.

Reply:
did research and fixed it lmao